homeassistant/components/restore_state.py

```diff
@@ -0,0 +1,86 @@
+"""
+Support for restoring entity states on startup.
+
+Component that records all events and state changes. Allows other components
+to query this database.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/recorder/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.core import HomeAssistant, split_entity_id
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_START, ATTR_RESTORED_STATE)
+from homeassistant.components.history import (
+    CONF_DOMAINS, CONF_EXCLUDE, CONF_ENTITIES, CONF_INCLUDE,
+    Filters, get_states, last_recorder_run)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType
+import homeassistant.util.dt as dt_util
+
+DOMAIN = 'restore_state'
+
+DEPENDENCES = ['recorder']
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: cv.FILTER_SCHEMA,
+}, extra=vol.ALLOW_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+_ALWAYS_OVERWRITE = (
```

Comment:
Me to have been watching this and trying to create something in the past but I guess everybody has different needs based on how you use HA.
In my case I don't want to restore all states as I use MQTT components that already restore states when HASS comes. The effect of restoring a MQTT light or switch based on previous state causing all kind of side effects.

The only thing I want to restore are input_* as those are quite often settings I like to retain.

So my idea would be to have an config option per component and not a component that restore all states.

Eg. with the input_select an option _restore_state: true_ (default to false)

Sound very much the solution that @balloob is proposing